collector/systemd_linux.go

```diff
@@ -57,19 +58,34 @@ func NewSystemdCollector() (Collector, error) {
 		"Whether the system is operational (see 'systemctl is-system-running')",
 		nil, nil,
 	)
+	summaryDesc := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, subsystem, "units"),
+		"Summary of systemd unit states", []string{"state"}, nil)
 	unitWhitelistPattern := regexp.MustCompile(fmt.Sprintf("^(?:%s)$", *unitWhitelist))
 	unitBlacklistPattern := regexp.MustCompile(fmt.Sprintf("^(?:%s)$", *unitBlacklist))
 
 	return &systemdCollector{
 		unitDesc:             unitDesc,
 		systemRunningDesc:    systemRunningDesc,
+		summaryDesc:          summaryDesc,
 		unitWhitelistPattern: unitWhitelistPattern,
 		unitBlacklistPattern: unitBlacklistPattern,
 	}, nil
 }
 
 func (c *systemdCollector) Update(ch chan<- prometheus.Metric) error {
-	units, err := c.listUnits()
+	allUnits, err := c.getAllUnits()
+	if err != nil {
+		return fmt.Errorf("couldn't get units: %s", err)
+	}
+
+	summary := summarizeUnits(allUnits)
+	if err != nil {
```

Comment:
This looks like leftover, `summarizeUnits()` doesn't return an `err`.

Reply:
Yep, there's another one for `filterUnits` as well a few lines after. Will clean both.